app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksActivity.java

```diff
@@ -15,36 +15,82 @@
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
 
 import com.example.bookwormadventuresdeluxe2.Utilities.UserCredentialAPI;
 import com.google.android.material.bottomnavigation.BottomNavigationView;
 import com.google.zxing.integration.android.IntentIntegrator;
 
 public class MyBooksActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener
 {
-
-    private BottomNavigationView navbar;
     private MyBooksFragment myBooksFragment = new MyBooksFragment();
     private SearchFragment searchFragment = new SearchFragment();
     private RequestsFragment requestsFragment = new RequestsFragment();
     private ProfileFragment profileFragment = new ProfileFragment();
+    private NotificationFragment notificationFragment = new NotificationFragment();
```

Comment:
We probably should not create a notification fragment here. Try this:

In MyBooksFragment.java inside onNotificationClick:
        NotificationFragment notificationFragment = new NotificationFragment();
        getFragmentManager().beginTransaction().add(R.id.frame_container, notificationFragment, "notificationFragment").hide(ActiveFragmentTracker.activeFragment).commit();

In NotificationFragment.java inside onBackClick:
        Fragment activeFragment = ActiveFragmentTracker.activeFragment;
        getFragmentManager().beginTransaction().hide(this).show(activeFragment).commit();

In MyBooksActivity.java inside onNavigationItemSelected:
/* If NotificationFragment is open, close it*/
        Fragment notificationFragment = fragmentManager.findFragmentByTag("notificationFragment");
        if( notificationFragment != null){
            fragmentManager.beginTransaction().remove(notificationFragment).commit();
        }

Reply:
Thanks for the suggestion, fixed